packages/services/api/src/modules/schema/providers/schema-publisher.ts

```diff
@@ -180,39 +216,13 @@ export class SchemaPublisher {
       latestSchemaVersion,
       latestComposableSchemaVersion,
     ] = await Promise.all([
-      this.targetManager.getTarget({
-        organization: input.organization,
-        project: input.project,
-        target: input.target,
-      }),
-      this.projectManager.getProject({
-        organization: input.organization,
-        project: input.project,
-      }),
-      this.organizationManager.getOrganization({
-        organization: input.organization,
-      }),
-      this.schemaManager.getLatestSchemas({
-        organization: input.organization,
-        project: input.project,
-        target: input.target,
-      }),
-      this.schemaManager.getLatestSchemas({
-        organization: input.organization,
-        project: input.project,
-        target: input.target,
-        onlyComposable: true,
-      }),
-      this.schemaManager.getMaybeLatestVersion({
-        organization: input.organization,
-        project: input.project,
-        target: input.target,
-      }),
-      this.schemaManager.getMaybeLatestValidVersion({
-        organization: input.organization,
-        project: input.project,
-        target: input.target,
-      }),
+      target$,
```

Comment:
@n1ru4l not sure you like it, but with this amount of function calls (and lines of code) it's hard connect a variable to a promise.